storage: panic in BenchmarkStoreRangeMerge

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf903c1]

goroutine 1610 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc4207fa000, 0x7f5f26eb94e8, 0xc420262b40)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:180 +0xb1
panic(0x1a9c280, 0x2957fb0)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
github.com/cockroachdb/cockroach/pkg/storage.(*NodeLiveness).Self(0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness.go:463 +0x41
github.com/cockroachdb/cockroach/pkg/storage.(*replicaGCQueue).shouldQueue(0xc420314a60, 0x7f5f26eb94e8, 0xc4206d1da0, 0x14d6611e9bb8a923, 0xc400000000, 0xc4206b5500, 0x0, 0x0, 0x0, 0xc42047e640, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_gc_queue.go:146 +0x341
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).MaybeAdd(0xc4205f2a00, 0xc4206b5500, 0x14d6611e9bb8a923, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:380 +0x235
github.com/cockroachdb/cockroach/pkg/storage.(*replicaScanner).waitAndProcess(0xc4204d52c0, 0x7f5f26eb94e8, 0xc420262b40, 0xed110f324, 0x337fa372, 0x2e18d60, 0xc420014700, 0xc4207fa000, 0xc4206b5500, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scanner.go:224 +0x30d
github.com/cockroachdb/cockroach/pkg/storage.(*replicaScanner).scanLoop.func1.1(0xc4206b5500, 0xc42066b440)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scanner.go:272 +0x90
github.com/cockroachdb/cockroach/pkg/storage.(*storeReplicaVisitor).Visit(0xc42066b440, 0xc42028b4f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:358 +0x1e6
github.com/cockroachdb/cockroach/pkg/storage.(*replicaScanner).scanLoop.func1(0x7f5f26eb94e8, 0xc420262b40)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scanner.go:274 +0x239
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc420312a50, 0xc4207fa000, 0xc420315300)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:195 +0xf7
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0xad
```

https://teamcity.cockroachdb.com/viewLog.html?tab=buildLog&logTab=tree&filter=debug&expand=all&buildId=309446&_focus=73146

Introduced in https://github.com/cockroachdb/cockroach/pull/17304.

--- a/pkg/storage/replica_gc_queue.go
+++ b/pkg/storage/replica_gc_queue.go
@@ -143,8 +143,11 @@ func (rgcq *replicaGCQueue) shouldQueue(
 		// we're decommissioning. If so, we should process the replica because it
 		// has probably already been removed from its raft group but doesn't know it.
 		// Without this, node decommissioning can stall on such dormant ranges.
-		if liveness, _ := repl.store.cfg.NodeLiveness.Self(); liveness != nil && liveness.Decommissioning {
-			return true, replicaGCPriorityDefault
+		// Make sure NodeLiveness isn't nil because it can be in tests/benchmarks.
+		if repl.store.cfg.NodeLiveness != nil {
+			if liveness, _ := repl.store.cfg.NodeLiveness.Self(); liveness != nil && liveness.Decommissioning {
+				return true, replicaGCPriorityDefault
+			}
 		}
 	}
 	return replicaGCShouldQueueImpl(now, lastCheck, lastActivity, isCandidate)

